JsonParam.mqh

```diff
@@ -0,0 +1,160 @@
+#define JSON_PARAM_MQH
+
+/**
+ * Enumeration.
+ */
+enum JsonParamType {
+  JsonParamBool,
+  JsonParamLong,
+  JsonParamDouble,
+  JsonParamString
+};
+
+class JsonNode;
+
+/**
+ * Key or value.
+ */
+class JsonParam
+{
+public:
+
+  /**
+   * Storing all integral values in a single union. We can't hold string here.
+   */
+  union Value {
+    bool _bool;
+    long _long;
+    double _double;
+  } _integral;
+  
+  string _string;
+  JsonParamType _type;
+  
+  /**
+   * Returns new JsonParam object from given source value.
+   */
+  static JsonParam* FromBool(long value) {
+    JsonParam* param = new JsonParam();
+    param._type = JsonParamBool;
+    param._integral._bool = value;
+    return param;
+  }
+
+  /**
+   * Returns new JsonParam object from given source value.
+   */
+  static JsonParam* FromLong(long value) {
+    JsonParam* param = new JsonParam();
+    param._type = JsonParamLong;
+    param._integral._long = value;
+    return param;
+  }
+  
+  /**
+   * Returns new JsonParam object from given source value.
+   */
+  static JsonParam* FromDouble(double value) {
+    JsonParam* param = new JsonParam();
+    param._type = JsonParamDouble;
+    param._integral._double = value;
+    return param;
+  }
+
+  /**
+   * Returns new JsonParam object from given source value.
+   */
+  static JsonParam* FromString (string &value) {
+    JsonParam* param = new JsonParam();
+    param._type = JsonParamString;
+    param._string = value;    
+    return param;
+  }
+  
+  /**
+   * Returns new JsonParam object from given source value.
+   */
+  static JsonParam* FromValue(bool value) {
+    return FromBool(value);
+  }
+
+  /**
+   * Returns new JsonParam object from given source value.
+   */
+  static JsonParam* FromValue(long value) {
+    return FromLong(value);
+  }
+
+  /**
+   * Returns new JsonParam object from given source value.
+   */
+  static JsonParam* FromValue(int value) {
+    return FromLong(value);
+  }
+
+  /**
+   * Returns new JsonParam object from given source value.
+   */
+  static JsonParam* FromValue(double value) {
+    return FromDouble(value);
+  }
+
+  /**
+   * Returns new JsonParam object from given source value.
+   */
+  static JsonParam* FromValue(string &value) {
+    return FromString(value);
+  }
+
+  /**
+   * Returns stringified version of the value. Note "forceQuotesOnString" flag.
+   */
+  string AsString(bool includeQuotes = false, bool forceQuotesOnString = true) {
+    switch (_type) {
+      case JsonParamBool:
+        return JSON::ValueToString(_integral._bool, includeQuotes);
+      case JsonParamLong:
+        return JSON::ValueToString(_integral._long, includeQuotes);
+      case JsonParamDouble:
+        return JSON::ValueToString(_integral._double, includeQuotes);
+      case JsonParamString:
+        return JSON::ValueToString(_string, includeQuotes || forceQuotesOnString);
+    }
+    
+    Print("JsonParam.AsString() called for an unknown value type: ", _type, "!");
```

Comment:
I think we should use either Log class/instance across the whole class (then e.g. `logger.Error(StringFormat("Some Error", _type));`) or wrap it into `__debug__`.

Reply:
Resolved by 7a3cdcbac7830595b7bfd95e0be5cbbb887a3ba8